src/main/java/org/opentripplanner/routing/algorithm/strategies/InterleavedBidirectionalHeuristic.java

```diff
@@ -296,15 +301,27 @@ public void doSomeWork() {
                         maxWeightSeen = weight;
                     }
                 }
-                continue;
+                if (!stopReached) {
+                    stopReached = true;
+                    rr.softWalkLimiting = false;
+                    rr.softPreTransitLimiting = false;
+                    if (s.walkDistance > rr.maxWalkDistance) {
+                        // Add 300 meters in order to search for nearby stops
+                        rr.maxWalkDistance = s.walkDistance + 300;
+                    }
+                }
+                if (!initialStop) continue;
             }
             // We don't test whether we're on an instanceof StreetVertex here because some other vertex types
             // (park and ride or bike rental related) that should also be explored and marked as usable.
             // Record the cost to reach this vertex.
-            if (!vertices.containsKey(v)) {
+            if (!vertices.containsKey(v) && !(v instanceof TransitStop)) {
                 vertices.put(v, (int) s.getWeight()); // FIXME time or weight? is RR using right mode?
             }
             for (Edge e : rr.arriveBy ? v.getIncoming() : v.getOutgoing()) {
+                if (v instanceof TransitStop && !(e instanceof StreetTransitLink)) {
```

Comment:
Perhaps also allow a StationStopEdge or a PathwayEdge?

Reply:
Line 322 is really just to not allow traversal from stops to their parent stops. It is unfortunate that "instance of" is just so heavily throughout the code, and it should probably be refactored at some point.

The search finds all stops in the vicinity by a street search. This works well for coordinate searches, but fails when searching from stop ids, because the search was originally made to terminate whenever it reaches a stop, so it just immediately terminates.

I have tried to allow it to continue searching down to the street network from the original stop, but then terminate when it reaches all subsequent stops.

Reply:
I was thinking of a case with pathways. For example, say TransitStop A is connected to TransitStop B via a PathwayEdge. B is really an entrance (location_type = 2 in stops.txt), it's connected to the street network, and A isn't. I think your fix handles the case where B is an endpoint, but not where A is. (There are some other issues with pathways so perhaps that case should be handled in a followup, but just pointing it out.)

Reply:
I see what you mean. We haven't used pathways at all, only routing by OSM network, so it is probably not supported with how it is written now.